diplom.py

```diff
@@ -18,17 +18,19 @@ def mean_average_precision(predicted_probability_distribution,
                            list_of_classes, 
                            test_y):
     total_error = 0
-    k = -1
-    for row in predicted_probability_distribution:
-        k += 1
-        classes_by_prob = []
+    for k,row in (enumerate (predicted_probability_distribution)):
+        prob_of_target_class = 0
+        place = 1
         for i in xrange(0, len(list_of_classes)):
```

Comment:
Since in next two loops you implicitly rely on the fact that `len(list_of_classes) == len(row)` I suggest to add an assertion before first for. I.e.:
`assert len(list_of_classes) == len(row)`